### PR TITLE
python: don't use optimizations on Darwin

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by FRidh** [comment]
Should fix https://github.com/NixOS/nixpkgs/issues/89912.

**Comment by FRidh** [comment]
@GrahamcOfBorg build python2 python3 python37

**Comment by vcunat** [comment]
The darwin failure seems unrelated (OOM killer, perhaps).  Let me try again:
@GrahamcOfBorg build python3

File: pkgs/development/interpreters/python/cpython/2.7/default.nix [hunk 1]
```diff
@@ -139,10 +140,11 @@ let
     '';
 
   configureFlags = [
-    "--enable-optimizations"
     "--enable-shared"
     "--with-threads"
     "--enable-unicode=ucs${toString ucsEncoding}"
+  ] ++ optionals enableOptimizations [
+    "--enable-optimizations"
```
Comment (vcunat) [inline pkgs/development/interpreters/python/cpython/2.7/default.nix:147]: We might prefer to avoid Linux rebuilds on staging-next by keeping the flag order for now.

**Comment by vcunat** [comment]
The borg build [failed even on the retry](https://github.com/NixOS/nixpkgs/runs/764969943).  I don't know how to best verify this.

**Comment by FRidh** [comment]
Just asked on nix-darwin if anyone could run
```
$ nix build -f https://github.com/FRidh/nixpkgs/archive/opt-darwin.tar.gz python python3 
```

**Comment by FRidh** [comment]
@GrahamcOfBorg build python3 python37

**Comment by FRidh** [comment]
Build succeeded according to our remote darwin builders over irc.
https://gist.github.com/cransom/9058b4e1b49ac13e62b223a42f8ab6d2
https://gist.github.com/cransom/b12e88534b5f6060f5292f6f0c4c94e3

**Comment by stolyaroleh** [comment]
Maybe a better idea would be to look at `stdenv.cc.isClang` instead? Otherwise, it will still attempt to build Python with optimizations when building with `clang` on Linux instead of `gcc`.
